jni/src/faiss_wrapper.cpp

```diff
@@ -491,6 +492,11 @@ std::unique_ptr<faiss::SearchParameters> buildSearchParams(const faiss::IndexIDM
         return hnswParams;
     }
 
+    auto nsgReader = dynamic_cast<const faiss::IndexNSG*>(indexReader->index);
+    if(nsgReader) {
+        // Search params not supported for the NSG index
+        return nullptr;
+    }
```

Comment:
is filtering not supported on NSG? if that is the case, then we need to throw the exception in the Query Builder. I cannot see how we are doing that.

Reply:
In code: https://github.com/facebookresearch/faiss/blob/3219e3d12e6fc36dfdfe17d4cf238ef70bf89568/faiss/IndexNSG.cpp#L81-L83

NSG Index do not support `SearchParameters` 
```
void IndexNSG::search(
        idx_t n,
        const float* x,
        idx_t k,
        float* distances,
        idx_t* labels,
        const SearchParameters* params) const {
    FAISS_THROW_IF_NOT_MSG(
            !params, "search params not supported for this index");
```

> then we need to throw the exception in the Query Builder

it is absolutely right, i'll add throw exception.

Reply:
https://github.com/opensearch-project/k-NN/pull/945/commits/819fcedce70fc452fb55152ab873a0e5dea53923 Fixed it https://github.com/opensearch-project/k-NN/blob/819fcedce70fc452fb55152ab873a0e5dea53923/jni/src/faiss_wrapper.cpp#L496-L499